6.analyze_features/scripts/1.anova.py

```diff
@@ -0,0 +1,245 @@
+df.head()
+
+
+# In[4]:
+
+
+# combine the genotype and idenity columns
+df["Metadata_genotype_side"] = df["Metadata_genotype"] + "_" + df["Metadata_side"]
+df["Metadata_genotype_identity_side"] = (
+    df["Metadata_genotype"] + "_" + df["Metadata_identity"] + "_" + df["Metadata_side"]
+)
+# split the features and the metadata
+metadata = df.columns.str.contains("Metadata")
+# filter the metadata
+metadata_df = df.loc[:, metadata]
+# filter the features
+features_df = df.loc[:, ~metadata]
+
+
+# ## Anova for genotype only
+
+# In[5]:
+
+
+# genotype_df = features_df.copy()
+# genotype_df.loc[:, "Metadata_genotype"] = metadata_df["Metadata_genotype"]
+# anova_results = anova_function(genotype_df, "Metadata_genotype")
+# # export the anova results
+# # out dir
+# out_dir = pathlib.Path("../../data/6.analysis_results/")
+# # create the dir if it does not exist
+# out_dir.mkdir(parents=True, exist_ok=True)
+# anova_results_path = out_dir / "anova_results_genotype.parquet"
+# anova_results.to_parquet(anova_results_path)
+
+
+# In[6]:
+
+
+# split the df into three genotypes
+high_df = df[df["Metadata_genotype"] == "high"]
+unsel_df = df[df["Metadata_genotype"] == "unsel"]
+wt_df = df[df["Metadata_genotype"] == "wt"]
+levene_test_results = {"feature": [], "levene_statistic": [], "levene_p_value": []}
+for feature in tqdm.tqdm(features_df.columns):
+    # calculate the levene test for each feature
+    levene_results = levene(wt_df[feature], unsel_df[feature], high_df[feature])
+    levene_test_results["feature"].append(feature)
+    levene_test_results["levene_statistic"].append(levene_results.statistic)
+    levene_test_results["levene_p_value"].append(levene_results.pvalue)
+
+levene_test_results_df = pd.DataFrame(levene_test_results)
+levene_test_results_df
+
+
+# ## Anova for genotype and side
+
+# In[7]:
+
+
+# genotype_df = features_df.copy()
+# genotype_df.loc[:, "Metadata_genotype_side"] = metadata_df["Metadata_genotype_side"]
+# anova_results = anova_function(genotype_df, "Metadata_genotype_side")
+# # export the anova results
+# # out dir
+# out_dir = pathlib.Path("../../data/6.analysis_results/")
+# # create the dir if it does not exist
+# out_dir.mkdir(parents=True, exist_ok=True)
+# anova_results_path = out_dir / "anova_results_genotype_side.parquet"
+# anova_results.to_parquet(anova_results_path)
+
+
+# ## Anova for genotype, side, and identity
+
+# In[8]:
+
+
+# genotype_df = features_df.copy()
+# genotype_df.loc[:, "Metadata_genotype_identity_side"] = metadata_df[
+#     "Metadata_genotype_identity_side"
+# ]
+
+# anova_results = anova_function(genotype_df, "Metadata_genotype_identity_side")
+
+# # export the anova results
+# # out dir
+# out_dir = pathlib.Path("../../data/6.analysis_results/")
+# # create the dir if it does not exist
+# out_dir.mkdir(parents=True, exist_ok=True)
+# anova_results_path = out_dir / "anova_results_genotype_side_identity.parquet"
+# anova_results.to_parquet(anova_results_path)
```

Comment:
Recommend removing commented-out code or uncommenting this out since it looks like this is important.